BedrockCore.cpp

```diff
@@ -145,37 +145,42 @@ bool BedrockCore::processCommand(BedrockCommand& command) {
             }
         }
     } catch (const SException& e) {
-        _handleCommandException(command, e.what(), true);
+        _handleCommandException(command, e, true);
     }
 
     // Done, return whether or not we need the parent to commit our transaction.
     command.complete = !needsCommit;
     return needsCommit;
 }
 
-void BedrockCore::_handleCommandException(BedrockCommand& command, const string& e, bool wasProcessing) {
+void BedrockCore::_handleCommandException(BedrockCommand& command, const SException& e, bool wasProcessing) {
     // If we were peeking, then we weren't in a transaction. But if we were processing, we need to roll it back.
     if (wasProcessing) {
         _db.rollback();
     }
-    const string& msg = "Error processing command '" + command.request.methodLine + "' (" + e + "), ignoring: " +
+    const string& msg = "Error processing command '" + command.request.methodLine + "' (" + e.what() + "), ignoring: " +
                         command.request.serialize();
-    if (SContains(e, "_ALERT_")) {
+    if (SContains(e.what(), "_ALERT_")) {
         SALERT(msg);
-    } else if (SContains(e, "_WARN_")) {
+    } else if (SContains(e.what(), "_WARN_")) {
         SWARN(msg);
-    } else if (SContains(e, "_HMMM_")) {
+    } else if (SContains(e.what(), "_HMMM_")) {
         SHMMM(msg);
-    } else if (SStartsWith(e, "50")) {
+    } else if (SStartsWith(e.what(), "50")) {
         SALERT(msg); // Alert on 500 level errors.
     } else {
         SINFO(msg);
     }
 
-    // If the command set a response before throwing an exception, we'll keep that as our response to use. Otherwise,
-    // we'll use the text of the error.
-    if (command.response.methodLine.empty()) {
-        command.response.methodLine = e;
+    // Set the response to the values from the exception, if set.
+    if (!e.method.empty()) {
```

Comment:
Hmmm, what about `If the command set a response before throwing an exception, we'll keep that as our response to use.` ?

Reply:
I looked at this. With the updated exceptions, if the exceptions sets headers, or a body, then that will replace the existing headers or body, but this isn't used anywhere (it will be used in Auth in the two places we used to use `additionalData`), so the headers or body wont get replaced here, if it was already set.

So, the only thing to consider then is the methodLine, but [we always set the methodLine to match the exception in Auth](https://github.com/Expensify/Server-Expensify/blob/master/auth/Auth.cpp#L840), so this will never change anything, as it's already been done in auth. The change in auth will simplify this so that we don't do this all twice, and we just let bedrock do it.

Reply:
Cool! 👍